src/main/java/com/kodilla/ecommercee/controller/CartController.java

```
@@ -1,71 +1,48 @@
 import java.util.List;
 
+
+@RequiredArgsConstructor
 @RestController
+@RequestMapping("/cart")
 @CrossOrigin("*")
-@RequestMapping("/carts")
 public class CartController {
+    private final CartService cartDbService;
+    private final ProductMapper productMapper;
 
-    List<CartDto> CartDtoList = new ArrayList<CartDto>(){
-        {
-            add(new CartDto(1L, "kurtka zimowa", "Pellentesque tempus interdum quam ut rhoncus. Donec...", new BigDecimal(100), 1L));
-            add(new CartDto(2L, "płaszcz", "Pellentesque tempus interdum quam ut rhoncus. Donec...", new BigDecimal(150), 1L));
-            add(new CartDto(8L,"krawat", "Pellentesque tempus interdum quam ut rhoncus. Donec...", new BigDecimal(50), 2L));
-
-        }
-    };
 
-    @GetMapping
-    public ResponseEntity<List<CartDto>> getCart(){
-        return ResponseEntity.ok(CartDtoList);
+    @PostMapping("{userId}")
+    public void createEmptyCart(@PathVariable("userId") Long userId) {
+        cartDbService.createEmptyCart(userId);
     }
 
-    @GetMapping(value = "{cartId}")
-    public ResponseEntity<CartDto> getCart(@PathVariable Long cartId){
-        for (CartDto cart : CartDtoList) {
-            if (cart.getId().equals(cartId)){
-                return ResponseEntity.ok(cart);
-            }
-        }
-        return ResponseEntity.notFound().build();
+    @GetMapping("{userId}")
+    public List<ProductDto> getCart(@PathVariable("userId") Long userId) {
+        return productMapper.mapToProductDtoList(cartDbService.getUserCart(userId));
     }
 
-    @DeleteMapping(value = "{cartId}")
-    public ResponseEntity<Void> deleteCart(@PathVariable Long cartId) {
-        for (CartDto cart : CartDtoList) {
-            if (cart.getId().equals(cartId)){
-                CartDtoList.remove(cart);
-                return ResponseEntity.ok().build();
-            }
-        }
-        return ResponseEntity.notFound().build();
+    @PutMapping("{userId}/{productId}")
```

I think we should throw an exception.

---

Same as above